app/src/main/java/ai/elimu/appstore/synchronization/AppListAdapter.java

```diff
@@ -438,6 +443,7 @@ public void run() {
                                 @Override
                                 public void onUninstallComplete(@NonNull String packageName) {
                                     notifyDataSetChanged();
+                                    holder.btnUninstall.setVisibility(View.GONE);
```

Comment:
Is it necessary to hide the button if list items with application status `DELETED` are hidden anyway?

Reply:
Basically once we complete uninstalling an application in our Android device, the Download button should appear, to allow user to re-download the app. Is that correct?
As I understand, the `DELETED` application status is another story, it's the status of the app in our web, unrelated to the application status (installed, downloaded, uninstalled) in our Android device.

Reply:
> Basically once we complete uninstalling an application in our Android device, the Download button should appear, to allow user to re-download the app. Is that correct?

No. Once an application has been uninstalled, the list item should be made invisible. An application with status 'DELETED' cannot be downloaded, and should not be visible in the list of apps.

Reply:
I see. So I've been misunderstanding the requirement of Uninstall function.
I'll update the code accordingly then.

Reply:
New implementation of the above logic has just been pushed.